test/com/google/enterprise/adaptor/fs/MockFileDelegate.java

```diff
@@ -189,30 +188,19 @@ public Path resolveDfsLink(Path doc) throws IOException {
     }
   }
 
-  private class MockDfsLinkDirectoryStream implements DirectoryStream<Path> {
-    private final List<Path> links;
-    private Iterator<Path> iterator;
-
+  private class MockDfsLinkDirectoryStream extends AbstractPathDirectoryStream {
     public MockDfsLinkDirectoryStream(MockFile namespace) throws IOException {
-      ImmutableList.Builder<Path> builder = ImmutableList.builder();
-      for (Path path : namespace.newDirectoryStream()) {
-        if (isDfsLink(path)) {
-          builder.add(path);
-        }
-      }
-      links = builder.build();
-    }
-
-    @Override
-    public Iterator<Path> iterator() {
-      Preconditions.checkState(iterator == null,
-          "DirectoryStream can only have one iterator.");
-      iterator = links.iterator();
-      return iterator;
-    }
-
-    @Override
-    public void close() {
+      paths = Iterables.filter(namespace.newDirectoryStream(),
```

Comment:
This is cute, but longer, harder to read, and inconsistent with MockDirectoryStream. The only benefit I can see is that it could be used as a direct argument to super(), if paths is made final in the superclass. Even there, a simple helper method or just removing this class and putting this code into newDfsLinkStream seems better.